contracts/depositor/CustomProxyFactory.sol

```diff
@@ -0,0 +1,40 @@
+pragma solidity ^0.6.0;
+
+// solium-disable security/no-inline-assembly
+// solium-disable security/no-low-level-calls
+contract CustomProxyFactory {
+
+  event ProxyCreated(address proxy);
+
+  function deployMinimal(address _logic, bytes memory _data) public returns (address proxy) {
+    bytes memory clone = deployCode(_logic);
+    assembly {
+      proxy := create(0, add(clone, 0x20), 0x37)
+    }
+
+    emit ProxyCreated(address(proxy));
+
+    if(_data.length > 0) {
+      (bool success,) = proxy.call(_data);
+      require(success, "constructor call failed");
```

Comment:
Use contract name in require statement as per convention: "CustomProxyFactory/constructor_call_failed"